### PR TITLE
feat(cli/fmt): support .markdown extension

### LINKED ISSUE #12194: Support .markdown for Markdown
The [IANA registration for the text/markdown media type](https://datatracker.ietf.org/doc/html/rfc7763#section-2) lists both `.md` and `.markdown` as file extensions. However, at present it looks like Deno (e.g. `deno fmt`) only recognizes `.md`. It ought to recognize both.

### PR DESCRIPTION
**Pull request by satyarohith**
Closes #12194

### CODE REVIEW

File: cli/tests/integration/fmt_tests.rs [hunk 1]
```diff
@@ -141,7 +141,7 @@ itest!(fmt_quiet_check_fmt_dir {
 });
 
 itest!(fmt_check_formatted_files {
-    args: "fmt --check fmt/regular/formatted1.js fmt/regular/formatted2.ts fmt/regular/formatted3.md fmt/regular/formatted4.jsonc",
+    args: "fmt --check fmt/regular/formatted1.js fmt/regular/formatted2.ts fmt/regular/formatted3.markdown fmt/regular/formatted4.jsonc",
```
Comment (satyarohith) [inline cli/tests/integration/fmt_tests.rs:144]: `.md` extension is exercised in `fmt_test()`

**Review by ahabhgk — APPROVED** [review]

**Comment by caspervonb** [comment]
Extensions as supported by GitHub:
- markdown
- mdown
- mkdn
- mkd
- md

Source: https://github.com/github/markup/blob/b865add2e053f8cea3d7f4d9dcba001bdfd78994/lib/github/markups.rb#L1

**Comment by kt3k** [comment]
Deno registry website does the similar thing to github https://github.com/denoland/dotland/blob/42c7fa426290ea5d7a37b9964d183be15ed700e2/util/registry_utils.ts#L273

**Comment by satyarohith** [comment]
> Do we stick to IANA only?

IANA seems the right thing, but considering GitHub and Registry, I included the extensions https://github.com/denoland/deno/pull/12195/commits/6da7e2da957125717dcc64768a6b9a926c66c34a

File: cli/tools/fmt.rs [hunk 2]
```diff
@@ -206,7 +206,7 @@ pub fn format_file(
   fmt_options: FmtOptionsConfig,
 ) -> Result<String, String> {
   let ext = get_extension(file_path).unwrap_or_else(String::new);
-  if ext == "md" {
+  if matches!(ext.as_str(), "md" | "markdown") {
```
Comment (lucacasonato) [inline cli/tools/fmt.rs:209]: The extra non IANA extensions are not mentioned here.

**Review by dsherret — APPROVED** [review]
LGTM. dprint-plugin-markdown also supports these same extensions.

**Pull request merged**